### PR TITLE
SALTO-6256: Implement Salesforce PermissionSet/MutingPermissionSet Elements Fixer

### PR DESCRIPTION
**Pull request by tamtamirr**
Implement Salesforce PermissionSet/MutingPermissionSet Elements Fixer.

---

- This PR is built on top the bug-fix from https://github.com/salto-io/salto/pull/6367

I combined the separated `permissionSet` and `profiles` handler to one. They were separated for a gradual rollout before.

---
_Release Notes_: 
_Salesforce Adapter_:
- Implement PermissionSet/MutingPermissionSet Elements fixer (turned off by default)

---
_User Notifications_: 
_None_

### CODE REVIEW

**Review by tomsellek — APPROVED** [review]

File: packages/salesforce-adapter/src/filters/field_references.ts [hunk 1]
```diff
@@ -207,13 +207,9 @@ export const addReferences = async (
...
   name: 'fieldReferencesFilter',
   onFetch: async elements => {
-    const typesToIgnore: string[] = []
-    if (!config.fetchProfile.isFeatureEnabled('generateRefsInProfiles')) {
-      typesToIgnore.push(PROFILE_METADATA_TYPE)
-    }
-    if (config.fetchProfile.isCustomReferencesHandlerEnabled('permisisonSets')) {
-      typesToIgnore.push(PERMISSION_SET_METADATA_TYPE, MUTING_PERMISSION_SET_METADATA_TYPE)
-    }
+    const typesToIgnore = config.fetchProfile.isCustomReferencesHandlerEnabled('profilesAndPermissionSets')
```
Comment (amitgoldarb) [inline packages/salesforce-adapter/src/filters/field_references.ts:210]: This effects current configs right? Just ensuring we don't have this turned off for some one currently

File: packages/salesforce-adapter/src/custom_references/handlers.ts [hunk 2]
```diff
@@ -26,26 +26,22 @@ import {
...
 
 const defaultCustomReferencesConfiguration: Required<CustomReferencesSettings> = {
-  profiles: true,
+  profilesAndPermissionSets: true,
   managedElements: true,
-  permisisonSets: true,
 }
 
 const defaultFixElementsConfiguration: Required<FixElementsSettings> = {
-  profiles: false,
```
Comment (amitgoldarb) [inline packages/salesforce-adapter/src/custom_references/handlers.ts:46]: Is removing this a breaking change in any way? Didn't test myself, just verifying we tested this manually

File: packages/salesforce-adapter/src/custom_references/handlers.ts [hunk 3]
```diff
@@ -26,26 +26,22 @@ import {
...
+  profilesAndPermissionSets: false,
```
Comment (amitgoldarb) [inline packages/salesforce-adapter/src/custom_references/handlers.ts:43]: This is good. Safe.

**Review by amitgoldarb — APPROVED** [review]
LGTM!
Approving to save cycles. Added some small comments

**Pull request merged**